src/views/AddItem.jsx

```diff
@@ -1,3 +1,108 @@
+import { useState, useEffect } from 'react';
+import { addItem } from '../api/firebase';
+
+const dayConverter = (text) => {
+	if (text === 'soon') {
+		return 7;
+	} else if (text === 'kind-of-soon') {
+		return 14;
+	} else {
+		return 30;
+	}
+};
 export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+	const [itemName, setItemName] = useState('');
+	const [frequency, setFrequency] = useState('soon');
+	const [message, setMessage] = useState('');
+	const listId = 'my test list';
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+		const daysUntilNextPurchase = dayConverter(frequency);
+		try {
+			await addItem(listId, { itemName, daysUntilNextPurchase });
+			setMessage(`${itemName} was added to the list`);
+			setItemName('');
+			setFrequency('soon');
+		} catch (err) {
+			console.error(err);
+			setMessage(`Failed to Add: ${itemName}`);
+		}
+	};
+	const handleChange = (e) => {
+		setFrequency(e.target.value);
+	};
+	useEffect(() => {
+		const handleKeyPress = async (e) => {
+			if (e.key === 'Enter' && e.keyCode === 13) {
+				e.preventDefault();
```

Comment:
**Suggestion:** Could `handleKeyPress()` and `handleSubmit()` be refactored to be more DRY?

It looks like rows 37-50 are using the same code as rows 19-29. If that's true, one idea for refactoring could be to call `handleSubmit()` on row 37, replacing the duplicate code there.

Reply:
At first, we used the handleSubmit function inside the handleKeyPress, but there were some issues getting the most updated state of itemName and frequency. That is why we separated them. 

I know it's repetitive as you mentioned. If you have a solution for this, we can update it accordingly.

Reply:
Oh interesting. Perhaps it has to do with `handleSubmit` being an async function, nested inside another async function? 🤔 I don't have any quick ideas for a solution. Since it's working as is and you already considered this, please don't let this suggestion block moving on and merging 🤪

Reply:
We actually removed the function. After further research and checking with Alejandro, we found out that our type=submit in the button was taking care of the enter key, so the function wasn't needed.